pkg/runtimeproxy/resexecutor/resource_executor.go

```diff
@@ -29,6 +29,7 @@ type RuntimeResourceExecutor interface {
 	UpdateResource(response interface{}) error
 	ResourceCheckPoint(response interface{}) error
 	DeleteCheckpointIfNeed(request interface{}) error
+	UpdateRequest(request interface{}) error
```

Comment:
UpdateRequest should be merged with UpdateResource, as they are both called after hook server response. we can call the new function UpdateRequest

Reply:
Done